internal/resource/config/datastore/jdbc_data_store.go

```diff
@@ -0,0 +1,386 @@
+				),
+			},
+		},
+		"blocking_timeout": schema.Int64Attribute{
+			Description: "The amount of time in milliseconds a request waits to get a connection from the connection pool before it fails. Omitting this attribute will set the value to the connection pool default.",
+			Computed:    true,
+			Optional:    true,
+			PlanModifiers: []planmodifier.Int64{
+				int64planmodifier.UseStateForUnknown(),
+			},
+		},
+		"idle_timeout": schema.Int64Attribute{
+			Description: "The length of time in minutes the connection can be idle in the pool before it is closed. Omitting this attribute will set the value to the connection pool default.",
+			Computed:    true,
+			Optional:    true,
+			PlanModifiers: []planmodifier.Int64{
+				int64planmodifier.UseStateForUnknown(),
+			},
+		},
+		"driver_class": schema.StringAttribute{
+			Description: "The name of the driver class used to communicate with the source database.",
+			Required:    true,
+		},
+		"connection_url": schema.StringAttribute{
+			Description: "The default location of the JDBC database. This field is required if no mapping for JDBC database location and tags are specified.",
+			Computed:    true,
+			Optional:    true,
+			Validators: []validator.String{
+				stringvalidator.AtLeastOneOf(
+					path.Expression.AtName(path.MatchRoot("jdbc_data_store"), "connection_url_tags"),
+					path.Expression.AtName(path.MatchRoot("jdbc_data_store"), "connection_url"),
+				),
+			},
+		},
+		"user_name": schema.StringAttribute{
+			Description: "The name that identifies the user when connecting to the database.",
+			Required:    true,
+		},
+		"allow_multi_value_attributes": schema.BoolAttribute{
+			Description: "Indicates that this data store can select more than one record from a column and return the results as a multi-value attribute.",
+			Computed:    true,
+			Optional:    true,
+			PlanModifiers: []planmodifier.Bool{
+				boolplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"validate_connection_sql": schema.StringAttribute{
+			Description: "A simple SQL statement used by PingFederate at runtime to verify that the database connection is still active and to reconnect if needed.",
+			Computed:    true,
+			Optional:    true,
+			Default:     stringdefault.StaticString(""),
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+	}
+
+	jdbcDataStoreSchema.Validators = []validator.Object{
+		objectvalidator.ExactlyOneOf(
+			path.MatchRelative().AtParent().AtName("custom_data_store"),
+			path.MatchRelative().AtParent().AtName("ldap_data_store"),
+			path.MatchRelative().AtParent().AtName("ping_one_ldap_gateway_data_store"),
+		),
+	}
+
+	return jdbcDataStoreSchema
+}
+
+func toStateJdbcDataStore(con context.Context, clientValue *client.DataStoreAggregation, plan dataStoreResourceModel) (types.Object, diag.Diagnostics) {
+	var allDiags, diags diag.Diagnostics
+	jdbcDataStore := *clientValue.JdbcDataStore
```

Comment:
Probably would be good to add a nil check here with a specific error diagnostic, so we don't have any chance of getting a panic. Same for the other data sources